src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -414,7 +419,7 @@ private void ProcessParameter(MshParameter p, PSObject inputObject, List<PSNoteP
                 }
             }
 
-            if (expressionResults.Count == 0)
+            if (expressionResults.Count == 0 && !ex.HasWildCardCharacters)
             {
                 //Commented out for bug 1107600
```

Comment:
And it would be better if you can add some comments here, explaining your purpose (allow `Select-Object -Property noexist-name` to return a `PSObject` with property `noexist-name`, unless `noexist-name` itself contains wildcards).
I find the implementation of `Select-Object` is so poorly commented (near to zero comment) that it becomes a pain to read the code. So let's add some when we make changes in this file.